spec/meilisearch/client/keys_spec.rb

```diff
@@ -56,7 +59,32 @@
       expect(key['description']).to eq('A new key to delete docs')
     end
 
-    it 'updates a key' do
+    it 'retrieves a list of keys' do
+      new_key = client.create_key(add_docs_key_options)
+
+      list = client.keys
+
+      expect(list.keys).to contain_exactly('limit', 'offset', 'results', 'total')
+      expect(list['results']).to eq([new_key])
+      expect(list['total']).to eq(1)
+    end
+
+    it 'paginates keys list with limit/offset' do
+      client.create_key(add_docs_key_options)
+
+      expect(client.keys(limit: 0, offset: 20)['results']).to be_empty
+      expect(client.keys(limit: 5, offset: 199)['results']).to be_empty
```

Comment:
I'm not sure I get the point of this test. Why do you set a limit of 5 but expect it to be empty?

Reply:
The answer is more complex than what I'm going to say to you, but what I can say is:

If we set a "random" limit/offset combination that will probably be out of bounds and make a request Meilisearch should return an empty array as the result.

I know it sounds counter-intuitive, but in the future our idea is to remove all of these tests and replace them by mocks like this:


```rb
    it 'sends limit/offset in the request' do
       expect(a_request(:get, "www.meilisearch.com").with(query: {"limit" => 5, "offset" => 10})).
  to have_been_made
  
       client.keys(limit: 5, offset: 10)
    end
```

Because if you agree with me, it will be enough to guarantee the contract between the API (Meili), and the caller (SDK).